backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import java.time.Clock;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+/** Class used to find a match in a list of Participants with the most recently added Participant */
+public final class FindMatchQuery {
+
+  /** Maximum difference in duration to be compatible */
+  private static int MAX_DURATION_DIFF = 15;
+  /** extra padding time to ensure large enough meeting time block */
+  private static int PADDING_TIME = 15;
+  /** Reference clock for "current" time */
```

Comment:
tiny nit: Now that it's a clock and not a ZoneDateTime, returning the "current time" is intrinsic to the type (clocks tell you what time it is right now) so this javadoc can be shortened or removed.